Added options to command find_package

bmarques1995 opened the pull request:
Now is possible to use the command find_package with cmake to find nghttp3

tatsuhiro-t commented:
@vszakats What do you think about this change?

vszakats commented on lib/CMakeLists.txt:
```diff
@@ -63,6 +63,33 @@ set(nghttp3_SOURCES
+  set(NGHTTP3_VERSION ${PROJECT_VERSION})
+  
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+      "${NGHTTP3_VERSION_CONFIG}" VERSION ${NGHTTP3_VERSION} COMPATIBILITY SameMajorVersion
+  )
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/config.cmake.in" "${NGHTTP3_PROJECT_CONFIG}" @ONLY)
+  
+  # Install cmake config files
+  install(
+      FILES "${NGHTTP3_PROJECT_CONFIG}" "${NGHTTP3_VERSION_CONFIG}"
+      DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
+
+   install(
+    EXPORT "${NGHTTP3_TARGETS_EXPORT_NAME}"
+    NAMESPACE "${NGHTTP3_NAMESPACE}"
+    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
```
```suggestion
  write_basic_package_version_file(
    "${NGHTTP3_VERSION_CONFIG}" VERSION ${NGHTTP3_VERSION} COMPATIBILITY SameMajorVersion
  )
  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/config.cmake.in" "${NGHTTP3_PROJECT_CONFIG}" @ONLY)
  
  # Install cmake config files
  install(
    FILES "${NGHTTP3_PROJECT_CONFIG}" "${NGHTTP3_VERSION_CONFIG}"
    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")

  install(
    EXPORT "${NGHTTP3_TARGETS_EXPORT_NAME}"
    NAMESPACE "${NGHTTP3_NAMESPACE}"
    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
```
nits: whitespace

vszakats commented on lib/CMakeLists.txt:
```diff
@@ -89,8 +117,8 @@ if(ENABLE_STATIC_LIB)
 
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libnghttp3.pc"
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+
```
```suggestion
```
nit: delete extra newline at the end of file.

vszakats commented on lib/cmake_modules/config.cmake.in:
```diff
@@ -0,0 +1,3 @@
+include(CMakeFindDependencyMacro)
+
+include("${CMAKE_CURRENT_LIST_DIR}/@NGHTTP3_TARGETS_EXPORT_NAME@.cmake")
```
A newline is missing at he end of this file.

What do you think of moving this file to the existing `cmake` directory instead?
Along with the generated `.cmake` modules above.

bmarques1995 replied:
It's okay for me

tatsuhiro-t commented:
@vszakats Thank you for the quick review!

bmarques1995 commented:
> @vszakats Thank you for the quick review!

I've modified the files, I've moved the pkgconfig to the end of the CMakeLists.txt, where it was located originally, about the modifications suggested by @vszakats, i perceived that the snippets indicated by him was at same identation level of the entry script, meaning without identation. I've moved the config.cmake.in to the same CMakeLists.txt dir.

vszakats commented:
Thanks @bmarques1995. Looks good to me, also agreeing on the `lib` directory for the new file.

tatsuhiro-t commented:
lib/config.cmake.in needs to be added to 
https://github.com/ngtcp2/nghttp3/blob/6bcfffb861501c2d1be02fc021e343ccb6456c74/lib/Makefile.am#L26 so that generated archive file contains it.

bmarques1995 commented:
Could u test again? If you approve the modifications i will squash the commits in one

tatsuhiro-t commented:
Thank you for PR.  Merged now.
